### ISSUE TITLE
Universe generation fails when python-fields is installed

### ISSUE THREAD

**Issue by Ram-Z** [freeorion/freeorion#2265]
### Environment

* **FreeOrion Version**: 0.4.8 and HEAD
* **Operating System**: Chakra Linux
* **Graphic card used**: GTX 760
* **Fetched as** 
  * Compiled from source

### Description

When starting a new game a message pops up saying something along the lines of "errors occured when generating the universe, gameplay might be impaired". There is then no systems/planets in the universe.

Looking at the logs I found:

```
23:45:04.279501 [debug] python : <string>:3 : Python current directory set to /home/ramsi/src/freeorion/default/python/universe_generation
23:45:04.283425 [error] python : configure_logging.py:197 : Uncaught exception: Traceback (most recent call last):
23:45:04.283491 [error] python : configure_logging.py:197 :   File "/home/ramsi/src/freeorion/build/default/python/universe_generation/universe_generator.py", line 14, in <module>
23:45:04.283503 [error] python : configure_logging.py:197 :     from fields import generate_fields
23:45:04.283514 [error] python : configure_logging.py:197 : ImportError: cannot import name generate_fields
23:45:04.283523 [error] python : configure_logging.py:197 : 
```

Only after looking at the `universe_generator.py` file and trying to jump to the `fields` import and landing on a file in `/usr/lib/python2/site-packages` that I realized what was going on. It's trying to load a function from a completely different module, [python-fields](https://github.com/ionelmc/python-fields).

### Expected Result

Universe is generated.

### Steps to reproduce

* `sudo pip install python-fields`
* Start game

**Comment by Dilvish-fo** [comment]
Hmm, I think this is arising primarily because when we initialize our boost python instance and add directories to the python system path, we append to the system path rather than prepend.  I suspect that just changing that would fix this problem, though we might also need to make a couple more of our python folders be formally packages (at least universe_generation), and it rather seems to me that we may as well do that with all of them for which it has not yet been done.   @Cjkjvfnby @Morlic-fo thoughts?

In the meantime, @Ram-Z, well hopefully you could do without the python-fields package for a while, or at least uninstall it right before running freeorion, or else you might be able to install an alternate installation of python 2.7, do not install python-fields to that one, and then execute freeorion via
```
PYTHONHOME=/path/to/alt/python ./freeorion
```

**Comment by Cjkjvfnby** [comment]
> Hmm, I think this is arising primarily because when we initialize our boost python instance and add directories to the python system path, we append to the system path rather than prepend. I suspect that just changing that would fix this problem,

Look like it should.

> though we might also need to make a couple more of our python folders be formally packages (at least universe_generation), and it rather seems to me that we may as well do that with all of them for which it has not yet been done

If the folder in python path it does not act like a package. `site-packages` folder does not have `__init__`.

**Comment by Ram-Z** [comment]
> In the meantime, @Ram-Z, well hopefully you could do without the python-fields package for a while

Yeah, absolutely. I have no idea why it was installed.

**Comment by o01eg** [comment]
> Hmm, I think this is arising primarily because when we initialize our boost python instance and add directories to the python system path, we append to the system path rather than prepend. I suspect that just changing that would fix this problem, though we might also need to make a couple more of our python folders be formally packages (at least universe_generation), and it rather seems to me that we may as well do that with all of them for which it has not yet been done.

Will it allows to use `site-packages` from freeorion python scripts if needed? I use `psycopg` on my server.

**Comment by Vezzra** [comment]
> Hmm, I think this is arising primarily because when we initialize our boost python instance and add directories to the python system path, we append to the system path rather than prepend. I suspect that just changing that would fix this problem

This is most likely correct.

That said, the real issue here (as @adrianbroher had pointed some time ago) is that we don't set up the FO Python modules (be it the ones interfacing with the C++ code or the utility/helper/etc. modules we provide) in a dedicated namespace.

All FO Python modules should have a prefix like `fo.*`, so we'd have `fo.ai.*` (or, more general, `fo.client.*`), `fo.common.*`, `fo.universe_generation.*` (or, more general, `fo.server.*`), etc.

That would prevent these kind of collisions.

**Comment by Dilvish-fo** [comment]
> Will it allows to use site-packages from freeorion python scripts if needed? I use psycopg on my server.

Yes, that will still be fine, we're not talking about removing anything from the pythonpath, just changing the search order.
